tests/out/hlsl/atomicSimple.hlsl

```diff
@@ -0,0 +1,40 @@
+
+groupshared uint foo;
+groupshared uint baz[2];
+
+[numthreads(2, 1, 1)]
+void corge(uint3 id : SV_GroupThreadID, uint3 __local_invocation_id : SV_GroupThreadID)
+{
+    if (all(__local_invocation_id == uint3(0u, 0u, 0u))) {
```

Comment:
question: What is `all(…)` here? 🤔

Reply:
Responding to my own question, looks like this is the [`all(…)` intrinsic](https://learn.microsoft.com/en-us/windows/win32/direct3dhlsl/dx-graphics-hlsl-all), which can be used on scalar, vectors, or matrices to determine that all components are non-zero. That…seems unnecessary here, but doesn't hurt anything, I suppose! Definitely nothing that should block here.

Reply:
`all` is required here since the `==` operator returns a boolean vector and the if statement expects a boolean (see https://learn.microsoft.com/en-us/windows/win32/direct3dhlsl/dx-graphics-hlsl-operators#comparison-operators).